[content collections] added typescript settings guidance via tabs (#2209)
withastro/docs: src/pages/en/guides/content-collections.md

});
```

### Update TypeScript configuration

> To benefit from the full TypeScript and autocompletion features of content collections, you may also need to update `tsconfig.json`. This is optional, but recommended.

Author: ```suggestion
To benefit from the full TypeScript and autocompletion features of content collections, you may also need to update `tsconfig.json`. Proper TypeScript configuration is required to [use schemas with your collections](en/guides/content-collections/#defining-a-collection-schema).
```

> To benefit from the full TypeScript and autocompletion features of content collections, you may also need to update `tsconfig.json`. Proper TypeScript configuration is required to [use schemas with your collections](en/guides/content-collections/#defining-a-collection-schema).

Author: ```suggestion
To benefit from the full TypeScript and autocompletion features of content collections, you may also need to update `tsconfig.json`. Proper TypeScript configuration is required to [use schemas with your collections](/en/guides/content-collections/#defining-a-collection-schema).
```

> To benefit from the full TypeScript and autocompletion features of content collections, you may also need to update `tsconfig.json`. Proper TypeScript configuration is required to [use schemas with your collections](/en/guides/content-collections/#defining-a-collection-schema).

Author: ```suggestion
To benefit from the full TypeScript and autocompletion features of content collections, you may also need to update `tsconfig.json`. Proper TypeScript configuration is required to [use schemas with your collections](#defining-a-collection-schema).
```

> To benefit from the full TypeScript and autocompletion features of content collections, you may also need to update `tsconfig.json`. Proper TypeScript configuration is required to [use schemas with your collections](#defining-a-collection-schema).
> 
> To enable the setting necessary for schemas, follow the instructions below based on your Astro project's current [TypeScript configuration](/en/guides/typescript/#setup). 

Reviewer: What do you think about condensing this? The stricter TS is specifically for schema-related features.

```suggestion
To benefit from the full TypeScript and autocompletion features of [using schemas with your collections](#defining-a-collection-schema), you may also need to update `tsconfig.json`. Follow the instructions below based on your Astro project's current [TypeScript configuration](/en/guides/typescript/#setup). 
```

<TypeScriptSettingTabs>
  <Fragment slot="relaxed">

>   If you are using Astro's `base` setting (e.g. you set up your project with "relaxed" TypeScript), add `"strictNullChecks": true` under `compilerOptions`. 

Reviewer: Small tweaks.

- Say “extending” `base` to reflect it’s literally in the `extends` option
- Tweaked the parenthetical to be clear “relaxed” refers to the `create astro` terminology

```suggestion
  If you are extending Astro's `base` TypeScript config (e.g. you chose "relaxed" TypeScript when creating your project with `create astro`), add `"strictNullChecks": true` under `compilerOptions`.
```

>   Or, you can change your `base` TypeScript setting to `strict` or `strictest` to use TypeScript's features throughout your entire Astro project.

Reviewer: This might suggest the `strictNullChecks` option _doesn’t_ apply throughout your project, so maybe safer to say something like this?

```suggestion
  Or, you can change your `base` TypeScript config to `strict` or `strictest` to use a broader set of stricter type-checks in your project.
```
